### PR TITLE
Skyhook columnar

### PR DESCRIPTION
**Pull request by ashayshirwadkar**
Added processing required for Apache Arrow

### CODE REVIEW

File: src/cls/tabular/cls_tabular.cc [hunk 1]
```diff
@@ -1517,7 +1504,24 @@ int exec_query_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
...
+                        }
+
+                        std::shared_ptr<arrow::Buffer> buffer;
+                        convert_arrow_to_buffer(table, &buffer);
+                        createFbMeta(meta_builder,
+                                     SFT_ARROW,
+                                     (unsigned char *)buffer->ToString().c_str(),
+                                     buffer->size());
+
+                        if (op.index_read)
```
Comment (jlefevre) [inline src/cls/tabular/cls_tabular.cc]: I think you can pull this out and put below the cases.

File: src/cls/tabular/cls_tabular.cc [hunk 2]
```diff
@@ -1936,37 +1948,43 @@ int transform_db_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
...
-    // TODO:remove
-    // get/set format type using fb_meta
-    /*
-    // Set the destination object format type
-    ret = set_sky_format_type(hctx, op.required_type);
-    if(ret < 0) {
-        CLS_ERR("transform_db_op: error setting sky_format_type entry to xattr %d", ret);
-        return ret;
+        // Write the object back to Ceph
+        ret = cls_cxx_write(hctx, offset, transformed_encoded_meta_bl.length(), &transformed_encoded_meta_bl);
```
Comment (jlefevre) [inline src/cls/tabular/cls_tabular.cc]: long line wrap.

File: src/cls/tabular/cls_tabular_utils.cc [hunk 3]
```diff
@@ -2343,9 +2379,8 @@ int read_from_file(const char *filename, std::shared_ptr<arrow::Buffer> *buffer)
 
 int write_to_file(const char *filename, arrow::Buffer* buffer)
```
Comment (jlefevre) [inline src/cls/tabular/cls_tabular_utils.cc:2378]: just make sure to add note this is just for testing arrow with local disk.

File: src/cls/tabular/cls_tabular_utils.cc [hunk 4]
```diff
@@ -3397,11 +3442,7 @@ int transform_arrow_to_fb(const char* data,
 
     // Placeholder function
     std::shared_ptr<arrow::Table> table;
-    std::shared_ptr<arrow::Buffer> buffer;
-
-    std::string str_data(data, data_size);
-    arrow::Buffer::FromString(str_data, &buffer);
-
+    std::shared_ptr<arrow::Buffer> buffer = arrow::MutableBuffer::Wrap(reinterpret_cast<uint8_t*>(const_cast<char*>(data)), data_size);
```
Comment (jlefevre) [inline src/cls/tabular/cls_tabular_utils.cc]: long line wrap.

**Review by jlefevre — CHANGES_REQUESTED** [review]
The re-build passed.  But let's still move the https and gnupg install lines up to here just for safety of install ordering.
https://github.com/uccross/skyhookdm-ceph/blob/skyhook-luminous/install-deps.sh#L120

File: src/cls/tabular/cls_tabular.cc (hunk 1 above)
Comment (ashayshirwadkar) [reply src/cls/tabular/cls_tabular.cc]: Done

**Review by jlefevre — APPROVED** [review]

**Pull request merged**